software/msisensor/msi/main.nf

```diff
@@ -0,0 +1,43 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process MSISENSOR_MSI {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::msisensor=0.5" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/msisensor:0.5--hb3646a4_2"
+    } else {
+        container "quay.io/biocontainers/msisensor:0.5--hb3646a4_2"
+    }
+
+    input:
+    tuple val(meta), path(normal_bam), path(normal_bai), path(tumor_bam), path(tumor_bai), val(metascan), path(homopolymers)
+
+    output:
+    tuple val(meta), path("*.output"), path("*.output_dis"), path("*.output_germline"), path("*.output_somatic"), emit: report
```

Comment:
this generates a tuple with `[val, path, path, path, path, path]`. 
If you want to only emit a single `report` rather than each output type in a separate channel, 
I'd suggest to use a wildcard to just generate a `[val, path]` tuple:

```
path("*.{output,output_dis,output_germline,output_somatic}")
```

Reply:
:+1: